### PR TITLE
4.1.x

### PR DESCRIPTION
**Pull request by espiegelberg**
Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

- [ ] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ ] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAGRAPH).
- [ ] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).
- [ ] You provide your full name and an email address registered with your GitHub account. If you’re a first-time submitter, make sure you have completed the [Contributor’s License Agreement form](https://support.springsource.com/spring_committer_signup).

### CODE REVIEW

**Comment by espiegelberg** [comment]
I am working on a Neo4j / SDN4 based project and am making use of Neo4jDataManipulationEvent’s. I have a use case where it would be helpful to know, as part of processing an AfterSaveEvent, if the entity was new or not before the save operation.

**Comment by mangrish** [comment]
@vince-bickers @luanne: Could you let me know if you want to add this feature and I can review and merge this.

**Comment by luanne** [comment]
I think the feature is useful, but, should this not be done in the OGM?

File: spring-data-neo4j/src/main/java/org/springframework/data/neo4j/util/EntityUtils.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	
+	/**
+	 * Determine if the domain entity is new, with "new" being defined as if the entity's GraphId property 
+	 * having a non-null value or not.
+	 * 
+	 * @param entity The domain entity.
+	 * @return Returns boolean true is the entity's GraphId property has a null value, boolean false if 
+	 * the GraphId property has a non-null value, and Boolean null if an access/argument exception 
+	 * occurred and/or if a qualifying GraphId property could not be found.
+	 */
```
Comment (luanne) [inline spring-data-neo4j/src/main/java/org/springframework/data/neo4j/util/EntityUtils.java]: The null value of the Boolean is a concern for me. If there's an exception or a missing GraphId, imho the exception should be thrown

**Comment by mangrish** [comment]
@luanne: It's for the `4.1.x` branch. It's not relevant in `master`/`4.2.x` as events are done differently.

**Comment by mangrish** [comment]
@espiegelberg: `4.1.x` is only for bug fixes. No new features are being added to this branch. `4.2.x` defines a brand new way to wire up events (see: https://github.com/neo4j-examples/sdn4-university/blob/spring-boot-1.5/src/main/java/school/Application.java).  That said; the OGM does not currently have a mechanism in the event code to determine if the object is new or not. 

If you would like to have this feature we can add it to the OGM. You will need to raise a ticket (or a PR if you are keen) on that project: https://github.com/neo4j/neo4j-ogm.

I'm going to close this PR but feel free to reference it in the OGM project if need be.

**Comment by espiegelberg** [comment]
You're approach of closing this item is perfectly reasonable.

First, and most importantly, is that, per @luanne's feedback, that the suggested feature is deemed a useful idea by the SDN team.

In hindsight though, it appears I should have proposed the feature before tearing off and implementing it as I would have learned that the 4.1.x branch is closed and that the 4.2 branch performs events different. I'll keep this in mind for future PR's.

@luanne's suggestion to implement the feature in the OGM is an improvement over my approach, so I think that's the way to move forward. I will open a feature request with the OGM project and, time allowing, pursue a PR (as I am making use of this feature on a current project).

**Pull request closed without merging**